test/TEST-41-ONESHOT-RESTART/testsuite.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+set -ex
+set -o pipefail
+
+systemd-analyze log-level debug
+systemd-analyze log-target console
+
+# These three commands should succeed.
+! systemd-run --unit=one -p Type=oneshot -p Restart=on-failure /bin/bash -c "exit 1"
```

Comment:
This isn't going to work, because this passes even without your changes:

```
% q systemd-run --unit=one -p Type=oneshot -p Restart=on-failure /bin/bash -c "exit 1" 
Failed to start transient service unit: Unit one.service has a bad unit file setting.
% systemctl show one.service -p NRestarts --value                                     
0
```